test/unit/features/segment_deletion_config_test.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Features::SegmentDeletionConfigTest < ActiveSupport::TestCase
+  def setup
+    @valid_config = {'email' => 'email@example.com', 'password' => 'example-password'}
+  end
+
+  attr_reader :valid_config
+
+  test 'marks as invalid if any value is missing' do
+    valid_config.each do |k, _v|
+      config = valid_config.dup
+      config.delete(k)
+      refute segment_deletion_config(config).enabled?
+    end
+  end
+
+  test 'enabled only in production and test environments' do
+    enabled_environments = %w[production test]
+    disabled_environments = available_environments - enabled_environments
+
+    enabled_environments.each do |current_environment|
+      stub_current_env(current_environment)
+      valid_config.each { |key, value| assert_equal value, segment_deletion_config.config.public_send(key) }
+      assert segment_deletion_config.enabled?
+    end
+
+    disabled_environments.each do |current_environment|
+      stub_current_env(current_environment)
+      valid_config.each { |key, value| assert_equal value, segment_deletion_config.config.public_send(key) }
+      refute segment_deletion_config.enabled?
+    end
+  end
+
+  private
+
+  def stub_current_env(current_env)
+    available_environments.each do |environment|
+      is_this_environment = (environment == current_env)
+      Rails.env.stubs("#{environment}?").returns(is_this_environment)
```

Comment:
Wow. This is dangerous! The whole test suite depends `Rails.env`. If something happens important in while stubbed we may perform something we really do not want to.

Reply:
We should just stub `valid_environment?`

Reply:
Ok, i will check this on monday 😄

Reply:
@guicassolato 
> Wow. This is dangerous! The whole test suite depends Rails.env. If something happens important in while stubbed we may perform something we really do not want to.

Ok, so I have to think what to do about this 🤔 


> We should just stub `valid_environment?`

I disagree 😝because then someone can add `development` or `preview` to that method and the test would pass, and it is important to make it fail in those cases for this: https://github.com/3scale/porta/pull/717/files#r268286677